### PR TITLE
use key-value format in stage0 file

### PR DESCRIPTION
**Pull request by onur-ozkan**
Currently, we are working on the python removal task on bootstrap. Which means we have to extract some data from the stage0 file using shell scripts. However, parsing values from the stage0.json file is painful because shell scripts don't have a built-in way to parse json files.

This change simplifies the stage0 file format to key-value pairs, which makes it easily readable from any environment.

See the zulip thread for more details: https://rust-lang.zulipchat.com/#narrow/stream/326414-t-infra.2Fbootstrap/topic/Using.20different.20format.20in.20the.20stage0.20file

### CODE REVIEW

**Comment by onur-ozkan** [comment]
r? Mark-Simulacrum

File: src/tools/build_helper/src/stage0_parser.rs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        if line.is_empty() {
+            continue;
+        }
+
+        // Ignore comments
+        if line.starts_with('#') {
+            continue;
+        }
+
+        let key_value: Vec<&str> = line.splitn(2, '=').collect();
```
Comment (Mark-Simulacrum) [inline src/tools/build_helper/src/stage0_parser.rs:44]: nit: [split_once](https://doc.rust-lang.org/nightly/std/primitive.str.html#method.split_once)?

File: src/tools/build_helper/src/stage0_parser.rs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            "compiler_date" => stage0.compiler.date = value.to_owned(),
+            "compiler_version" => stage0.compiler.version = value.to_owned(),
+
+            "rustfmt_date" => {
+                let mut rustfmt = stage0.rustfmt.unwrap_or(VersionMetadata::default());
+                rustfmt.date = value.to_owned();
+                stage0.rustfmt = Some(rustfmt);
+            }
+            "rustfmt_version" => {
+                let mut rustfmt = stage0.rustfmt.unwrap_or(VersionMetadata::default());
```
Comment (Mark-Simulacrum) [inline src/tools/build_helper/src/stage0_parser.rs:64]: https://doc.rust-lang.org/nightly/std/option/enum.Option.html#method.get_or_insert?

**Review by Mark-Simulacrum — COMMENTED** [review]
r=me

**Pull request merged**